add RSC docs (#517)
vercel/swr-site: pages/docs/with-nextjs.en-US.mdx

# Usage with Next.js

## App Router [#app-dir]

### Server Components [#rsc]

<Callout type="default" emoji="✅">
   In Next.js App Router, all components are React Server Components (RSC) by default. **You could only import the key serialization APIs from SWR in RSC.**
</Callout>

```tsx filename="app/page.tsx" copy

> import { unstable_serialize } from 'swr' // ✅ works 
> import { unstable_serialize as infinite_unstable_serialize } from 'swr/infinite' // ✅ works 

Reviewer: ```suggestion
import { unstable_serialize } from 'swr' // ✅ Available in server components 
import { unstable_serialize as infinite_unstable_serialize } from 'swr/infinite' // ✅ Available in server components
```

```

<Callout type="error">
   You could not import any other APIs from SWR since they are not available in RSC.
</Callout>

```tsx filename="app/page.tsx" highlight={1}

> import useSWR from 'swr' // ❌ it will throw an error

Reviewer: ```suggestion
import useSWR from 'swr' // ❌ This is not allowed in server components
```

```

### Client Components

> You need to mark your components with `'use client'` directive so that SWR could work properly.

Reviewer: ```suggestion
You can mark your components with `'use client'` directive or import SWR from client components, both ways will allow you to use the SWR client data fetching hooks
```

```tsx filename="app/page.tsx" highlight={1} copy
'use client'
import useSWR from 'swr'
export default Page() {
  const { data } = useSWR('/api/user', fetcher)
  return <h1>{data.name}</h1>
}
```

> If you need to use `SWRConfig` to do some global setup in your `layout` or `page` component, but you also don't want to make them client components. You could create a separate provider component.

Reviewer: ```suggestion
If you need to use `SWRConfig` to confifgure global settings in server components `layout` or `page`, creating a separate provider client component to setup the provider and configuration then use it in the server component pages.
```

Together with SWR, you can pre-render the page for SEO, and also have features such as caching, revalidation, focus tracking, refetching on interval on the client side.

> You can use the `fallback` option of [`SWRConfig`](/docs/global-configuration) to pass the pre-fetched data as the initial value of all SWR hooks.

Reviewer: ```suggestion
You can use the `fallback` option of [`SWRConfig`](/docs/global-configuration) to pass the pre-fetched data as the initial value of all SWR hooks. 
```
